yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,221 @@
+            video_id, expected_status=200).get('did')
+
+        getcomments = self.get_param('getcomments', False)
+        meta = self._download_json(
+            'https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread',
+            video_id, headers={'Content-Type': 'application/json'},
+            query={'uri': f'at://{did}/app.bsky.feed.post/{video_id}',
+                   'depth': 1000 if getcomments else 0,
+                   'parentHeight': 1000 if getcomments else 0},
+            expected_status=200).get('thread')
+        post, record_embed = meta.get('post'), traverse_obj(meta, ('post', 'record', 'embed'))
+
+        formats, subs = self._extract_m3u8_formats_and_subtitles(
+            traverse_obj(post, ('embed', 'playlist')),
+            video_id, 'mp4', 'm3u8_native', m3u8_id='hls', fatal=False,
+            note='Downloading HD m3u8 information', errnote='Unable to download HD m3u8 information')
+        blob_cid = traverse_obj(post, ('embed', 'cid'), ('record', 'embed', 'video', 'ref', '$link'))
+        formats.append({
+            'format_id': 'blob',
+            'url': f'https://bsky.social/xrpc/com.atproto.sync.getBlob?did={did}&cid={blob_cid}',
+            'ext': mimetype2ext(traverse_obj(record_embed, ('video', 'mimeType')), 'mp4'),
+            'width': traverse_obj(record_embed, ('aspectRatio', 'width'), expected_type=int_or_none),
+            'height': traverse_obj(record_embed, ('aspectRatio', 'height'), expected_type=int_or_none),
+            'filesize': traverse_obj(record_embed, ('video', 'size'), expected_type=int_or_none),
+        })
+
+        handle = traverse_obj(post, ('author', 'handle'))
+        uploader = traverse_obj(post, ('author', 'displayName'))
+        description = traverse_obj(post, ('record', 'text'))
+        extractor = self.extract_comments(meta)
+
+        return {
+            'id': video_id,
+            'title': f'{uploader}: "{description}"',
+            'formats': formats,
+            'description': description,
+            'thumbnail': traverse_obj(post, ('embed', 'thumbnail'), expected_type=url_or_none),
+            'alt_title': traverse_obj(post, ('embed', 'alt'), ('record', 'embed', 'alt')),
+            'uploader': uploader,
+            'channel': handle,
+            'uploader_id': did,
+            'channel_id': did,
+            'uploader_url': f'https://bsky.app/profile/{handle}',
+            'channel_url': f'https://bsky.app/profile/{did}',
+            'timestamp': parse_iso8601(traverse_obj(post, ('record', 'createdAt'))),
+            'like_count': post.get('likeCount'),
+            'repost_count': post.get('repostCount'),
+            'comment_count': post.get('replyCount'),
+            'webpage_url': url,
+            'tags': post.get('labels', []) + traverse_obj(post, ('record', 'langs'), default=[]),
+            'comments': [] if not extractor else [*(extractor().get('comments'))[0]][1:],
```

Comment:
```suggestion
            '__post_extractor': self.extract_comments(meta),
```
not sure if this is what you're trying to do(the suggestions on Line 157 and Line 198 must be applied to make this work)
Please also remember to update tests, if necessary

Reply:
Thank you - this is much better :)